### PR TITLE
Make a page of product details by clicking on the product

### PR DESCRIPTION
**Pull request by YehorAltShuler**
https://www.figma.com/file/qMnX1YMS15Hu3vXTSvlPVH/Coffee-Delivery-App-Exploration-(Community)?node-id=1%3A2&t=KyRzNP9sccntBmMZ-0

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -11,20 +14,45 @@ class MyApp extends StatelessWidget {
...
-        scaffoldBackgroundColor: AppColors.PRIMARY_COLOR,
-        colorScheme: ColorScheme.fromSwatch().copyWith(
-          primary: AppColors.PRIMARY_COLOR,
-          secondary: AppColors.SECONDARY_COLOR,
-        ),
-        appBarTheme: const AppBarTheme(
-          color: AppColors.PRIMARY_COLOR,
-          elevation: 0,
+    return ChangeNotifierProvider(
+      create: (BuildContext context) => Products(),
```
Comment (ldemyanenko) [inline lib/main.dart:18]: Yes, usually you initiate global app state providers in this place

File: lib/screens/ProductDetails/product_details_screen.dart [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import 'package:readmore/readmore.dart';
+
+class ProductDetailsScreen extends StatelessWidget {
+  const ProductDetailsScreen({super.key});
+
+  static const routeName = '/product-details-screen';
+
+  @override
+  Widget build(BuildContext context) {
+    final product = ModalRoute.of(context)?.settings.arguments as Product;
```
Comment (ldemyanenko) [inline lib/screens/ProductDetails/product_details_screen.dart:17]: Taking arguments in this place will force this operation on every build. Where did you find this practice? @YehorAltShuler

File: lib/screens/ProductDetails/product_details_screen.dart [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                  child: ClipRRect(
+                    borderRadius: BorderRadius.circular(50),
+                    child: BackdropFilter(
+                      filter: ImageFilter.blur(sigmaX: 10, sigmaY: 10),
+                      child: Material(
+                        color: Colors.transparent,
+                        child: IconButton(
+                          onPressed: Navigator.of(context).pop,
+                          icon: const Icon(
+                            Icons.arrow_back,
```
Comment (ldemyanenko) [inline lib/screens/ProductDetails/product_details_screen.dart:50]: You should use asset from design. This arrow is different. Same for other material icons

File: lib/screens/ProductDetails/product_details_screen.dart [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+                            product.rate.toString(),
+                            style: Theme.of(context).textTheme.labelMedium,
+                          )
+                        ],
+                      ),
+                    ),
+                  ],
+                ),
+                SizedBox(height: AppConstants.kDefaultPadding.toDouble()),
+                ReadMoreText(
```
Comment (ldemyanenko) [inline lib/screens/ProductDetails/product_details_screen.dart:107]: nice

File: lib/widgets/ProductCard/card_product_add.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        color: AppColors.VERY_FADED_SECONDARY_COLOR,
+        borderRadius: BorderRadius.circular(10),
+      ),
+      child: Row(
+        children: [
+          Expanded(
+            child: Center(
+              child: Text(
+                price.toString(),
+                style: Theme.of(context)
```
Comment (ldemyanenko) [inline lib/widgets/ProductCard/card_product_add.dart:23]: I would suggest ass font family to the Theme definition. I doubt that the same TextStyle is used in design with different font families

File: lib/widgets/input_chips.dart [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+
+class InputChip extends StatefulWidget {
```
Comment (ldemyanenko) [inline lib/widgets/input_chips.dart:3]: Is not used in project

**Review by ldemyanenko — CHANGES_REQUESTED** [review]

File: lib/screens/ProductDetails/product_details_screen.dart (hunk 2 above)
Comment (YehorAltShuler) [reply lib/screens/ProductDetails/product_details_screen.dart:17]: Uuuh... Just saw this everywhere in tutorials

File: lib/screens/ProductDetails/product_details_screen.dart (hunk 4 above)
Comment (YehorAltShuler) [reply lib/screens/ProductDetails/product_details_screen.dart:107]: +

File: lib/widgets/input_chips.dart (hunk 6 above)
Comment (YehorAltShuler) [reply lib/widgets/input_chips.dart:3]: I deleted it

File: lib/screens/ProductDetails/product_details_screen.dart (hunk 3 above)
Comment (YehorAltShuler) [reply lib/screens/ProductDetails/product_details_screen.dart]: +

File: lib/widgets/ProductCard/card_product_add.dart (hunk 5 above)
Comment (YehorAltShuler) [reply lib/widgets/ProductCard/card_product_add.dart]: there is 2 font families in dsign definitely.

Comment (ldemyanenko) [reply lib/widgets/ProductCard/card_product_add.dart]: Yes, but are they used in same TextStyle combinations? Anyway it's doesn't looks good to define theme in this way. Try defining them on app theme level

File: lib/screens/ProductDetails/product_details_screen.dart (hunk 2 above)
Comment (ldemyanenko) [reply lib/screens/ProductDetails/product_details_screen.dart:18]: Looks like Flutter team doesn't mind such an approach https://docs.flutter.dev/cookbook/navigation/navigate-with-arguments

File: lib/widgets/ProductCard/card_product_add.dart (hunk 5 above)
Comment (YehorAltShuler) [reply lib/widgets/ProductCard/card_product_add.dart]: You was right, it seems like Open sans uses for body Text

File: lib/main.dart [hunk 7]
```diff
@@ -11,20 +14,47 @@ class MyApp extends StatelessWidget {
...
+          fontFamily: 'Rosarivo',
+          elevatedButtonTheme: ElevatedButtonThemeData(
+              style: ElevatedButton.styleFrom(
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(10.0),
+            ),
+            backgroundColor: AppColors.SECONDARY_COLOR,
+          )),
+          textTheme: ThemeData().textTheme.copyWith(
+                headlineSmall: const TextStyle(color: Colors.white),
```
Comment (ldemyanenko) [inline lib/main.dart:31]: nice

**Review by ldemyanenko — APPROVED** [review]

**Pull request merged**